# 1141_社會資料分析 加分題
學生：劉晏成

In [1]:
from load import load_sav
import pandas as pd

df = load_sav("../data/brilliant_working.sav")

In [2]:
variable_value_labels = df.attrs['variable_value_labels']
column_names = df.attrs['column_names']
column_names_to_labels = df.attrs['column_names_to_labels']

## Q1: 按照下列規則，將「配偶的教育程度」 （a10）重新編碼成「配偶教育年數」 （s_eduy）  後，報告「配偶教育年數」的平均數和標準差。

| 教育程度類別                 | 教育年數 |
|------------------------|------|
| 無、自修                   | 0    |
| 小學肄業                   | 6    |
| 國（初）中、初職               | 9    |
| 高中普通科、高中職業科、高職、士官學校    | 12   |
| 五專、二專、軍警專修班、軍警專科班、空中行專 | 14   |
| 三專                     | 15   |
| 空中大學、軍警官學校、技術學院、科大、大學  | 16   |
| 碩士                     | 19   |
| 博士                     | 24   |


In [5]:
df_1 = df[['a10']].copy()

In [17]:
spouse_education_variable_value_labels = variable_value_labels['a10']
spouse_education_variable_value_labels

{1.0: '無(不識字)',
 2.0: '自修(識字、私塾)',
 3.0: '小學肄業',
 4.0: '國(初)中',
 5.0: '初職',
 6.0: '高中普通科',
 7.0: '高中職業科',
 8.0: '高職',
 9.0: '士官學校',
 10.0: '五專',
 11.0: '二專',
 12.0: '三專',
 13.0: '軍警專修班',
 14.0: '軍警專科班',
 15.0: '空中行專',
 16.0: '空中大學',
 17.0: '軍警官學校',
 18.0: '技術學院、科大',
 19.0: '大學',
 20.0: '碩士',
 21.0: '博士',
 22.0: '其他',
 997.0: '拒答',
 998.0: '漏答',
 999.0: '不適用'}

依據編碼表編碼並預覽類別數目

In [19]:
import numpy as np

def categorize_education(value):
    category_mapping = {
        1: 0,
        2: 0,
        3: 6,
        4: 9,
        5: 9,
        6: 12,
        7: 12,
        8: 12,
        9: 12,
        10: 14,
        11: 14,
        13: 14,
        14: 14,
        15: 14,
        12: 15,
        16: 16,
        17: 16,
        18: 16,
        19: 16,
        20: 19,
        21: 24
    }
    
    return category_mapping.get(value, np.nan)

df_1_recoded = df_1.copy()
df_1_recoded.loc[:, 's_eduy'] = df_1_recoded['a10'].apply(categorize_education)

df_1_recoded['s_eduy'].value_counts().sort_index()

s_eduy
0.0       8
6.0      55
9.0      88
12.0    204
14.0     80
15.0      6
16.0    145
19.0     31
24.0      7
Name: count, dtype: int64

In [25]:
df_1_describe = df_1_recoded['s_eduy'].describe()

mean = df_1_describe['mean']
std = df_1_describe['std']
median = df_1_describe['50%']
_min = df_1_describe['min']
_max = df_1_describe['max']
q1 = df_1_describe['25%']
q3 = df_1_describe['75%']
mode = df_1_recoded.loc[:, 's_eduy'].mode()[0]
interquartile_range = q3 - q1

In [26]:
import pandas as pd

answer = pd.DataFrame({
    'Value': [mean, median, mode, _max - _min, q1, q3, interquartile_range, std]
}, index = ['Mean', 'Median', 'Mode', 'Range', 'Q1', 'Q3', 'Interquartile Range', 'Standard Deviation'])
answer

Value
Mean                 12.591346
Median               12.000000
Mode                 12.000000
Range                24.000000
Q1                   12.000000
Q3                   16.000000
Interquartile Range   4.000000
Standard Deviation    3.785330

### Answer: 

配偶教育年數的平均數和標準差分別為12.59 與3.79

## Q2 比較不同性別者（a1）的配偶教育年數是否有顯著差異？
1. 列出不同性別者的配偶教育年數平均分數
2. 按照假設檢定四步驟，做獨立樣本 t 檢定

In [33]:
df_2 = new_df = pd.concat([df['a1'], df_1_recoded['s_eduy']], axis=1).copy()

df_2

a1  s_eduy
0    2.0    16.0
1    1.0    12.0
2    2.0     NaN
3    1.0     NaN
4    2.0    16.0
..   ...     ...
863  2.0     6.0
864  2.0    12.0
865  2.0     9.0
866  1.0     NaN
867  1.0     6.0

[868 rows x 2 columns]

## Q3: 比較不同年齡分組者（agegroup）的配偶教育年數是否有顯著差異？
1. 列出不同年齡分組者的配偶教育年數平均分數
2. 按照假設檢定四步驟，做變異數分析的整體檢定
3. 如果變異數分析的整體檢定達到顯著水準，接著做事後比較

In [35]:
df_3 = new_df = pd.concat([df['agegroup'], df_1_recoded['s_eduy']], axis=1).copy()

df_3

agegroup  s_eduy
0         1.0    16.0
1         1.0    12.0
2         1.0     NaN
3         1.0     NaN
4         3.0    16.0
..        ...     ...
863       3.0     6.0
864       2.0    12.0
865       3.0     9.0
866       2.0     NaN
867       3.0     6.0

[868 rows x 2 columns]